### PR TITLE
Fixed MakeFieldReadonlyCodeFixProvider for volatile fields.

### LINKED ISSUE #42759: IDE0044 makes volatile fields readonly, resulting in errors
**Version Used**: 
C# Tools   3.6.0-2.20167.1+926207070b0c62d6a76aec4627ec26dee17870bb

**Steps to Reproduce**:
```C#
class Program
{
    private volatile object _obj;

    static void Main() { }
}
```

**Expected Behavior**:
IDE0044 doesn't fire.

**Actual Behavior**:
IDE0044 fires, and running it (e.g. on the whole solution) changes the code to be:
```C#
class Program
{
    private readonly volatile object _obj;

    static void Main() { }
}
```
which results in the error:
```
CS0678	'Program._obj': a field cannot be both volatile and readonly
```

### PR DESCRIPTION
**Pull request by zaytsev-victor**
Fixes #42759

### CODE REVIEW

File: src/Analyzers/Core/CodeFixes/MakeFieldReadonly/AbstractMakeFieldReadonlyCodeFixProvider.cs [hunk 1]
```diff
@@ -67,7 +67,9 @@ private async Task MakeFieldReadonlyAsync(Document document, SyntaxEditor editor
 
                 if (declarationDeclarators.Count == fieldDeclarators.Count())
                 {
-                    editor.SetModifiers(fieldDeclarators.Key, editor.Generator.GetModifiers(fieldDeclarators.Key) | DeclarationModifiers.ReadOnly);
+                    var modifiers = editor.Generator.GetModifiers(fieldDeclarators.Key) - DeclarationModifiers.Volatile | DeclarationModifiers.ReadOnly;
```
Comment (CyrusNajmabadi) [inline src/Analyzers/Core/CodeFixes/MakeFieldReadonly/AbstractMakeFieldReadonlyCodeFixProvider.cs:70]: 1. using `-` here is very uncommon.  Generally to wipe the bits here it's preferred to do `x & ~y`.  
2. this pattern is repeated twice.  can you extract a helper for this?

Comment (zaytsev-victor) [reply src/Analyzers/Core/CodeFixes/MakeFieldReadonly/AbstractMakeFieldReadonlyCodeFixProvider.cs]: This isn't an Enum. ```DeclarationModifiers``` doesn't have override for ```~``` operator.

File: src/Analyzers/Core/CodeFixes/MakeFieldReadonly/AbstractMakeFieldReadonlyCodeFixProvider.cs [hunk 2]
```diff
@@ -96,6 +98,9 @@ private async Task MakeFieldReadonlyAsync(Document document, SyntaxEditor editor
             }
         }
 
+        private static DeclarationModifiers WithReadOnly(DeclarationModifiers modifiers)
+            => modifiers - DeclarationModifiers.Volatile | DeclarationModifiers.ReadOnly;
```
Comment (CyrusNajmabadi) [inline src/Analyzers/Core/CodeFixes/MakeFieldReadonly/AbstractMakeFieldReadonlyCodeFixProvider.cs:102]:
```suggestion
            => (modifiers - DeclarationModifiers.Volatile) | DeclarationModifiers.ReadOnly;
```

Comment (CyrusNajmabadi) [reply src/Analyzers/Core/CodeFixes/MakeFieldReadonly/AbstractMakeFieldReadonlyCodeFixProvider.cs:102]: or, my preference:
```suggestion
            => modifiers.WithIsVolatile(false).WithReadOnly(true);
```

**Pull request merged**